tests/test_atom_mapping.py

```diff
@@ -988,3 +988,21 @@ def test_min_threshold():
 
     with pytest.raises(NoMappingError, match="Unable to find mapping with at least 18 atoms"):
         atom_mapping.get_cores(mol_a, mol_b, **core_kwargs, max_visits=10000)
+
+
+def test_get_cores_and_diagnostics():
+    mols = read_sdf(hif2a_set)
+    n_pairs = 30
+    random_pair_idxs = np.random.default_rng(2024).choice(len(mols), size=(n_pairs, 2))
+
+    for i_a, i_b in random_pair_idxs:
+        mol_a = mols[i_a]
+        mol_b = mols[i_b]
+        all_cores, diagnostics = atom_mapping.get_cores_and_diagnostics(mol_a, mol_b, **DEFAULT_ATOM_MAPPING_KWARGS)
+
+        assert len(all_cores) > 0
+        assert all(len(core) == len(all_cores[0]) for core in all_cores)
+
+        assert diagnostics.core_size >= len(all_cores[0])
+        assert diagnostics.num_cores >= len(all_cores)
+        assert diagnostics.total_nodes_visited >= diagnostics.core_size
```

Comment:
nit: add a few-word comment about this comparison? wasn't immediately clear to me, but I think it makes sense (have to visit at least one search node per atom-pair added to core)

Reply:
Added in https://github.com/proteneer/timemachine/pull/1231/commits/1ab494c6d3f40c0932c9ff3802a6aec31bbf5abe

Reply:
Whoops, your original wording was more accurate than what I added. Fixed in https://github.com/proteneer/timemachine/pull/1231/commits/6a5b88609c189705e093bd5d362a96e785fdc26b